cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/OperationHttpHandler.java

```diff
@@ -98,4 +176,115 @@ public void failOperation(FullHttpRequest request, HttpResponder responder,
   }
+
+  private ScanOperationRunsRequest getScanRequest(
+      String namespaceId, String pageToken, Integer pageSize, String filter) {
+    ScanOperationRunsRequest.Builder builder = ScanOperationRunsRequest.builder();
+    builder.setNamespace(namespaceId);
+    if (pageSize != null) {
+      builder.setLimit(pageSize);
+    }
+    if (pageToken != null) {
+      builder.setScanAfter(pageToken);
+    }
+    if (filter != null && !filter.isEmpty()) {
+      OperationRunFilter operationRunFilter = getFilter(filter);
+      builder.setFilter(operationRunFilter);
+    }
+    return builder.build();
+  }
+
+  private OperationRunFilter getFilter(String filter) {
+    Map<String, String> filterKeyValMap = parseKeyValStr(filter, FILTER_SPLITTER);
+    OperationType operationType = null;
+    OperationRunStatus operationStatus = null;
+
+    for (Map.Entry<String, String> entry : filterKeyValMap.entrySet()) {
+      String filterValue = entry.getValue();
+      OperationFilterKey filterKey = OperationFilterKey.valueOf(entry.getKey());
+
+      switch (filterKey) {
+        case TYPE:
+          try {
+            operationType = OperationType.valueOf(filterValue.toUpperCase());
+          } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Invalid OperationType: " + filterValue);
+          }
+          break;
+        case STATUS:
+          try {
+            operationStatus = OperationRunStatus.valueOf(filterValue);
+          } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Invalid OperationRunStatus: " + filterValue);
+          }
+          break;
+        default:
+          throw new IllegalArgumentException("Unknown filter key: " + filterKey);
+      }
+    }
+    return new OperationRunFilter(operationType, operationStatus);
+  }
+
+ /**
+ * Parses a string containing key-value pairs separated by a specified splitter.
+ *
+ * @param input    The input string containing key-value pairs.
+ * @param splitter The string used to split key-value pairs.
+ * @return A {@code Map<String, String>} containing the parsed key-value pairs.
+ * @throws IllegalArgumentException If the input does not match the expected key=val pair pattern.
+ */
+
+  private static Map<String, String> parseKeyValStr(String input, String splitter) {
+    Map<String, String> keyValMap = new HashMap<>();
+    String[] keyValPairs = input.split(splitter);
+  
+    for (String keyValPair : keyValPairs) {
+      Matcher matcher = KEY_VALUE_PATTERN.matcher(keyValPair.trim());
+      
+      if (matcher.matches()) {
+          keyValMap.put(matcher.group(1).trim().toUpperCase(), matcher.group(2).trim().toUpperCase());
+      } else {
+          throw new IllegalArgumentException("Invalid filter key=val pair: " + keyValPair);
+      }
+  }
+    return keyValMap;
+  }
+
+  private void validateNamespace(@Nullable String namespace) throws BadRequestException, NamespaceNotFoundException, AccessException{
```

Comment:
I dont think this can be shorthened to 

```
  private NamespaceId validateNamespaceId(String namespaceId) throws BadRequestException {
    try {
      return new NamespaceId(namespaceId);
    } catch (IllegalArgumentException e) {
      throw new BadRequestException(e.getMessage(), e);
    }
  }
```

see `SourceControlManagementHttpHandler` for reference